code/game/gamemodes/umbrage/umbrage_objects.dm

```diff
@@ -0,0 +1,156 @@
+	<span class='danger'>You begin siphoning [L]'s mental energy...</span>")
+	L << "<span class='userdanger'><i>AAAAAAAAAAAAAA-</i></span>"
+	L.Stun(3)
+	playsound(L, 'sound/magic/devour_will.ogg', 100, 0) //T A S T Y   S O U L S
+	if(!do_mob(user, L, 30))
+		user.Weaken(3)
+		L << "<span class='boldwarning'>All right. You're all right.</span>"
+		L.Weaken(3)
+		qdel(src)
+		return
+	user.visible_message("<span class='warning'>[user] gently lowers [L] to the ground...</span>", "<span class='velvet bold'>...aranupdejc</span><br>\
+	<span class='boldnotice'>You devour [L]'s will. Your psi has been fully restored.\n\
+	Additionally, you have gained one lucidity. Use it to purchase and upgrade abilities.</span><br>\
+	<span class='warning'>[L] is now severely weakened and will take some time to recover.</span>")
+	playsound(L, 'sound/magic/devour_will_victim.ogg', 50, 0)
+	U.psi = U.max_psi
+	U.lucidity++
+	U.lucidity_drained++
+	U.drained_minds += L.mind
+	L << "<span class='userdanger'>You suddenly feel... empty. Thoughts try to form, but flit away. You slip into a deep, deep slumber...</span>"
+	L << sound('sound/magic/devour_will_end.ogg', volume = 75)
+	linked_ability.victims += L
+	L.Paralyse(30)
+	L.stuttering += 40
+	L.confused += 40
+	L.reagents.add_reagent("zombiepowder", 2) //Brief window of vulnerability to veiling
+	qdel(src)
+	#warn Change this dark bead recovery timer - 2 minutes, maybe?
+	spawn(30) //I don't like to use a spawn here, but because of how it works I have to
+		if(linked_ability && L)
+			linked_ability.victims -= L
+			user << "<span class='notice'>[L] has recovered from their draining and is vulnerable to Devour Will again.</span>"
+	return 1
+
+
+
+//Umbral tendrils: Formed by the Pass ability. See umbrage_abilities.dm for more details.
+/obj/item/weapon/umbral_tendrils
+	name = "umbral tendrils"
+	desc = "A mass of purple, glowing tentacles."
+	icon_state = "dark_bead"
+	flags = NODROP
+	resistance_flags = FIRE_PROOF | LAVA_PROOF | UNACIDABLE | INDESTRUCTIBLE
+	var/datum/umbrage/linked_umbrage
+
+
+//Psionic barrier: Created during Divulge. Has a regenerating health pool and protects the umbrage from harm.
+/obj/structure/psionic_barrier
+	name = "psionic barrier"
+	desc = "A violet tint to the air. It doesn't seem to have a physical presence."
+	obj_integrity = 200
+	max_integrity = 200
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "purplesparkles"
+	resistance_flags = FIRE_PROOF | LAVA_PROOF | UNACIDABLE
+	anchored = 1
+	opacity = 0
+	density = 1
+	luminosity = 1
+	mouse_opacity = 2
+
+/obj/structure/psionic_barrier/New()
+	..()
+	START_PROCESSING(SSprocessing, src)
+	QDEL_IN(src, 500)
+
+/obj/structure/psionic_barrier/Destroy()
+	if(!obj_integrity)
+		visible_message("<span class='warning'>[src] vanishes in a burst of violet energy!</span>")
+		playsound(src, 'sound/effects/hit_on_shattered_glass.ogg', 50, 1)
+		PoolOrNew(/obj/effect/overlay/temp/revenant/cracks, get_turf(src))
```

Comment:
In this master branch, it's there, so I'll have to purge it when I fix conflicts at the end